### DISCUSSION TITLE
How to remove black whitespace ?

### DISCUSSION THREAD

**Discussion by dimaslanjaka** [ZFC-Digital/puppeteer-real-browser#186] [Q&A] [answered]
Look at right half screen, it blank whitespace. How to remove ?

![image](https://github.com/user-attachments/assets/ad596e02-9f60-42c9-8e07-72b2b309fb15)

here my codes

```js
import path from "path";
import { connect } from "puppeteer-real-browser";
import { handleExitWithoutRestart } from "../process.js";

const start = async () => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const { page, browser } = await connect({
    headless: false,
    ignoreHTTPSErrors: true,
    customConfig: { userDataDir: path.resolve("assets/chrome-profiles/default") },
    args: [
      "--disable-blink-features=AutomationControlled", // Disables Blink feature: Automation Controlled.
      "--disable-features=HeavyAdIntervention", // Disable Chrome's blocking of intrusive ads
      "--disable-features=AdInterestGroupAPI", // Prevents blocking based on ad interest group
      "--disable-popup-blocking", // Disable pop-up blocking
      "--no-default-browser-check",
      "--no-first-run",
      "--ignore-certificate-errors",
      "--hide-crash-restore-bubble",
      "--autoplay-policy=no-user-gesture-required",
      "--window-size=1380,800",
      "--disable-legacy-window", // Disables legacy window support.
      "--disable-gpu", // Disables the GPU hardware acceleration.
      "--disable-dev-shm-usage", // Disables shared memory usage.
      "--disable-features=IsolateOrigins,site-per-process",
      "--blink-settings=imagesEnabled=true",
      "--no-sandbox",
      "--disable-setuid-sandbox"
    ]
  });
  await page.goto("https://google.com", { timeout: 0 });
};

handleExitWithoutRestart();
start();
```

**Comment by mdervisaygan** [comment, marked as answer]
> Look at right half screen, it blank whitespace. How to remove ?
> 
> ![image](https://private-user-images.githubusercontent.com/12471057/372322913-ad596e02-9f60-42c9-8e07-72b2b309fb15.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc3NTkxMzksIm5iZiI6MTcyNzc1ODgzOSwicGF0aCI6Ii8xMjQ3MTA1Ny8zNzIzMjI5MTMtYWQ1OTZlMDItOWY2MC00MmM5LThlMDctNzJiMmIzMDlmYjE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAxVDA1MDAzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiOTQ1NGI0MGVlZDY3MjMwM2NhYzFjN2NmYjYzNjAwNWQ1ZjliN2RlZDhiNGFlOWI4Y2IxNTM0ODUwYjcwNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ESEpH4zGBHskyZHkEyZyr55Ft0veEMgNEkNi6wM1Ny4)
> 
> here my codes
> 
> ```js
> import path from "path";
> import { connect } from "puppeteer-real-browser";
> import { handleExitWithoutRestart } from "../process.js";
> 
> const start = async () => {
>   // eslint-disable-next-line @typescript-eslint/no-unused-vars
>   const { page, browser } = await connect({
>     headless: false,
>     ignoreHTTPSErrors: true,
>     customConfig: { userDataDir: path.resolve("assets/chrome-profiles/default") },
>     args: [
>       "--disable-blink-features=AutomationControlled", // Disables Blink feature: Automation Controlled.
>       "--disable-features=HeavyAdIntervention", // Disable Chrome's blocking of intrusive ads
>       "--disable-features=AdInterestGroupAPI", // Prevents blocking based on ad interest group
>       "--disable-popup-blocking", // Disable pop-up blocking
>       "--no-default-browser-check",
>       "--no-first-run",
>       "--ignore-certificate-errors",
>       "--hide-crash-restore-bubble",
>       "--autoplay-policy=no-user-gesture-required",
>       "--window-size=1380,800",
>       "--disable-legacy-window", // Disables legacy window support.
>       "--disable-gpu", // Disables the GPU hardware acceleration.
>       "--disable-dev-shm-usage", // Disables shared memory usage.
>       "--disable-features=IsolateOrigins,site-per-process",
>       "--blink-settings=imagesEnabled=true",
>       "--no-sandbox",
>       "--disable-setuid-sandbox"
>     ]
>   });
>   await page.goto("https://google.com", { timeout: 0 });
> };
> 
> handleExitWithoutRestart();
> start();
> ```

https://github.com/zfcsoftware/puppeteer-real-browser?tab=readme-ov-file#pagesetviewport-method-is-not-working-what-should-i-do

```
const start = async () => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const { page, browser } = await connect({
        headless: false,
        ignoreHTTPSErrors: true,
        customConfig: { userDataDir: path.resolve("assets/chrome-profiles/default") },
        args: [
            "--disable-blink-features=AutomationControlled", // Disables Blink feature: Automation Controlled.
            "--disable-features=HeavyAdIntervention", // Disable Chrome's blocking of intrusive ads
            "--disable-features=AdInterestGroupAPI", // Prevents blocking based on ad interest group
            "--disable-popup-blocking", // Disable pop-up blocking
            "--no-default-browser-check",
            "--no-first-run",
            "--ignore-certificate-errors",
            "--hide-crash-restore-bubble",
            "--autoplay-policy=no-user-gesture-required",
            "--window-size=1380,800",
            "--disable-legacy-window", // Disables legacy window support.
            "--disable-gpu", // Disables the GPU hardware acceleration.
            "--disable-dev-shm-usage", // Disables shared memory usage.
            "--disable-features=IsolateOrigins,site-per-process",
            "--blink-settings=imagesEnabled=true",
            "--no-sandbox",
            "--disable-setuid-sandbox"
        ],
        connectOption: {
            defaultViewport: null
        },
    });
    await page.goto("https://google.com", { timeout: 0 });
};

handleExitWithoutRestart();
start();
```
